src/coreComponents/physicsSolvers/fluidFlow/kernels/compositional/IHUPhaseFlux.hpp

```diff
@@ -366,11 +364,11 @@ computeFractionalFlowViscous( localIndex const numPhase,
 
     for( localIndex ke = 0; ke < numFluxSupportPoints; ++ke )
     {
-      dFractionalFlow_dP[ke] -= fractionalFlow * dTotMob_dP[k_up_ppu] / LvArray::math::max( totMob, minTotMob );
+      dFractionalFlow_dP[ke] -= fractionalFlow * dTotMob_dP[ke] / LvArray::math::max( totMob, minTotMob );
```

Comment:
`dTotMob_dP` can have contributions from both cells, right?
so here we are collecting those contributions into `dFractionalFlow_dP`
using `k_up_ppu` does not make much sense, no?

Reply:
I think unless you'd been careful when building `dTotMob_dP` both are not quite right isn't it ?

Then as the formula is:

```
d(f)_L = ... - f_HU d(mobT_PPU)d(XL)/mobT_PPU
d(f)_R = ... - f_HU d(mobT_PPU)d(XR)/mobT_PPU
```
so if counter current flow occurs,
```
mobT_PPU = mob_a_L + mob_b_R
```
so you don't want 
```
d(mobT)d(XL) = d(mob_a_L + mob_b_L)d(XL)
```
and rather 
```
d(mobT_PPU) = d(mob_a_L )d(XL)
```
no ?

Reply:
thanks for the explanation
the logic is like that:
`dTotMob_dP` is an array of 2 elements
when being assembled it will write into `k_up`, can be different for each phase
after that `dTotMob_dP[0]` contains derivatives of totMob wrt pressure in cell 0 and `dTotMob_dP[1]` contains derivatives of totMob wrt pressure in cell 1
we are taking `dFractionalFlow_dP` derivative wrt pressure in cell 0/1 by including contribution from `dTotMob_dP` wrt to pressure in cell 0/1
there will be no `d(mob_b_L)d(XL)` contribution in your example, it should effectively become
```
d(mobT)d(XL) = d(mob_a_L)d(XL)
d(mobT)d(XR) = d(mob_b_R)d(XR)
```
please let me know if something still does not match

Reply:
No,I think it matches then. Devils in details with those interpolation scheme :+1: